pkg/signup/service/signup_service_test.go

```diff
@@ -1889,6 +1910,139 @@ func (s *TestSignupServiceSuite) TestIsPhoneVerificationRequired() {
 
 }
 
+func (s *TestSignupServiceSuite) TestGetSignupUpdatesUserSignupAnnotations() {
+
+	s.ServiceConfiguration(configuration.Namespace(), false, "", 5)
+
+	// Create a new UserSignup, set its UserID and AccountID annotations
+	userSignup := s.newUserSignupComplete()
+
+	err := s.FakeUserSignupClient.Tracker.Add(userSignup)
+	require.NoError(s.T(), err)
+
+	mur := &toolchainv1alpha1.MasterUserRecord{
+		TypeMeta: v1.TypeMeta{},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      userSignup.Status.CompliantUsername,
+			Namespace: configuration.Namespace(),
+		},
+		Spec: toolchainv1alpha1.MasterUserRecordSpec{
+			UserAccounts: []toolchainv1alpha1.UserAccountEmbedded{{TargetCluster: "member-123"}},
+		},
+		Status: toolchainv1alpha1.MasterUserRecordStatus{
+			Conditions: []toolchainv1alpha1.Condition{
+				{
+					Type:   toolchainv1alpha1.MasterUserRecordReady,
+					Status: "true",
+				},
+			},
+		},
+	}
+	err = s.FakeMasterUserRecordClient.Tracker.Add(mur)
+	require.NoError(s.T(), err)
+
+	s.Run("confirm nothing changed when context nil", func() {
+		_, err := s.Application.SignupService().GetSignup(nil, userSignup.Name, userSignup.Spec.Username)
+		require.NoError(s.T(), err)
+
+		modified, err := s.FakeUserSignupClient.Get(userSignup.Name)
+		require.NoError(s.T(), err)
+
+		require.NotContains(s.T(), modified.Annotations, toolchainv1alpha1.SSOUserIDAnnotationKey)
```

Comment:
Can we test it with some annotation present? So we can make sure the annotations are not actually changed instead of being deleted.

Reply:
I've changed the test ordering so that the nil context test now occurs after one of the annotations are actually set, to confirm that it's not changed.